src/platform/forms-system/src/js/components/FieldTemplate.jsx

```diff
@@ -2,12 +2,22 @@ import React from 'react';
 import classNames from 'classnames';
 import get from '../../../../utilities/data/get';
 import { isReactComponent } from '../../../../utilities/ui';
-// import environment from 'platform/utilities/environment';
+
+export function checkIsTouched(formContext, currentFieldId) {
+  return (
+    formContext.touched[currentFieldId] ||
+    Object.keys(formContext.touched).some(touchedFieldId =>
+      // if "root_field" is touched, then we consider
+      // "root_field_child" as also touched, but
+      // "root_fieldSibling" should not be.
+      currentFieldId.match(new RegExp(`^${touchedFieldId}(?=$|[^a-zA-Z0-9])`)),
```

Comment:
additional context on this regex:
$ = end of line, so currentFieldId === touchFieldId
[^a-zA-Z0-9], the next character is not continuing the same string, so if our next character is "_", ":", "-" for example that is considered matching. couldn't use "/w" because that includes "_"